File: benchmarks/bench_strings.py
result = ""
for i in range(500):
    result = result + str(i) + " "
assert len(result) > 0

words = result.strip().split(" ")
assert len(words) == 500

upper_count = 0
for w in words:
    u = w.upper()
    if u == w:
        upper_count += 1

print(f"processed {len(words)} words")

File: benchmarks/bench_strings.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	result := ""
	for i := 0; i < 500; i++ {
		result = result + strconv.Itoa(i) + " "
	}
	if len(result) == 0 {
		panic("empty")
	}
	words := strings.Split(strings.TrimSpace(result), " ")
	if len(words) != 500 {
		panic("count")
	}
	upperCount := 0
	for _, w := range words {
		if strings.ToUpper(w) == w {
			upperCount++
		}
	}
	fmt.Println("processed", len(words), "words")
}
